sources/kube.go

```diff
@@ -259,6 +272,7 @@ func newKubeSource() (*KubeSource, error) {
 		client:      kubeClient,
 		lastQuery:   time.Now(),
 		kubeletPort: *argKubeletPort,
+		activePods:  make(map[PodInstance]bool),
```

Comment:
Yeah. we can do that.

On Wed, Jan 21, 2015 at 3:21 PM, Rohit Jnagal <notifications@github.com>
wrote:

> In sources/kube.go
> <https://github.com/GoogleCloudPlatform/heapster/pull/69#discussion_r23343819>
> :
>
> > @@ -259,6 +272,7 @@ func newKubeSource() (*KubeSource, error) {
> >  		client:      kubeClient,
> >  		lastQuery:   time.Now(),
> >  		kubeletPort: *argKubeletPort,
> > +		activePods:  make(map[PodInstance]bool),
>
> yes, this is a problem because we don't keep a list of pods being tracked.
> Would it be good enough to regenerate the list in getPods() - purge all old
> entries that are not present in the new listing from getPods()?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/heapster/pull/69/files#r23343819>.
>